apps/game_backend/lib/game_backend/units/characters.ex

```diff
@@ -214,4 +216,72 @@ defmodule GameBackend.Units.Characters do
     |> Skin.changeset(attrs)
     |> Repo.insert()
   end
+
+  @doc """
+  Gets a Skin.
+  ## Examples
+      iex> get_skin_by_name(id)
+      {:ok, %Skin{}}
+      iex> get_skin_by_name(wrong_id)
+      {:error, :not_found}
+  """
+  def get_skin_by_name(skin_name) do
+    Repo.get_by(Skin, name: skin_name)
+    |> case do
+      nil -> {:error, :not_found}
+      skin -> {:ok, skin}
+    end
+  end
+
+  def get_skin_purchase_cost_by_currency(skin, currency_id) do
+    purchase_cost =
+      Map.get(skin, :purchase_costs)
+      |> Enum.find(fn purchase_cost -> purchase_cost.currency_id == currency_id end)
+
+    case purchase_cost do
+      nil -> {:error, :not_found}
+      purchase_cost -> {:ok, purchase_cost}
+    end
+  end
+
+  @doc """
+  Receives a user_id, a skin_id and a list of CurrencyCosts.
+  Inserts new UserSkin from given Skin for given User.
+  Substract the amount of Currency to User by given params.
+  Returns {:ok, map_of_ran_operations} in case of success.
+  Returns {:error, failed_operation, failed_value, changes_so_far} if one of the operations fail.
+  """
+  def buy_skin(%{user_id: user_id, skin_id: skin_id, unit_id: unit_id}, purchase_costs_list) do
+    curse_id = GameBackend.Utils.get_game_id(:curse_of_mirra)
+
+    Multi.new()
+    |> Multi.run(:unit_skin, fn _, _ -> insert_unit_skin(%{user_id: user_id, skin_id: skin_id, unit_id: unit_id}) end)
+    |> Multi.run(:currencies, fn _, _ -> Currencies.substract_currencies(user_id, purchase_costs_list) end)
+    |> Multi.run(:updated_user, fn _, _ -> GameBackend.Users.get_user_by_id_and_game_id(user_id, curse_id) end)
+    |> Repo.transaction()
+  end
```

Comment:
we should ensure the user does not already have the skin, I was able to double buy Kenzu's skin

Reply:
You're totally right. Done!